IdentityCore/src/oauth2/aad_base/MSIDAADTokenResponse.m

```diff
@@ -79,6 +79,11 @@ - (NSString *)accountIdentifier
     return self.clientInfo.accountIdentifier;
 }
 
+- (NSString *)accountUpn
+{
+    return [super accountUpn] ?: self.additionalUserId;
```

Comment:
I assume you're setting additionalUserId property on the broker side?

Reply:
correct